clients/src/main/java/org/apache/kafka/common/requests/ListOffsetsRequest.java

```diff
@@ -58,17 +58,21 @@ public class ListOffsetsRequest extends AbstractRequest {
     public static class Builder extends AbstractRequest.Builder<ListOffsetsRequest> {
         private final ListOffsetsRequestData data;
 
-        public static Builder forReplica(short allowedVersion, int replicaId) {
-            return new Builder((short) 0, allowedVersion, replicaId, IsolationLevel.READ_UNCOMMITTED);
+        public static Builder forConsumer(boolean requireTimestamp,
+                                          IsolationLevel isolationLevel) {
+            return forConsumer(requireTimestamp, isolationLevel, false, false, false);
         }
 
         public static Builder forConsumer(boolean requireTimestamp,
                                           IsolationLevel isolationLevel,
                                           boolean requireMaxTimestamp,
+                                          boolean requireEarliestTimestamp,
                                           boolean requireTieredStorageTimestamp) {
             short minVersion = 0;
             if (requireTieredStorageTimestamp)
                 minVersion = 9;
+            else if (requireEarliestTimestamp)
```

Comment:
@FrankYang0529 Could you please take a look?

Reply:
Hi @chia7712 and @frankvicky, What does `requireEarliestTimestamp` mean? `EARLIEST_TIMESTAMP` or `EARLIEST_LOCAL_TIMESTAMP`? If it's `EARLIEST_LOCAL_TIMESTAMP`, probably we should change the parameter name as `requireEarliestLocalTimestamp`, so users don't misunderstand it.

Reply:
Sure, I will update it